File: leet_3162_count_good.cc
class Solution {
public:
    int numberOfPairs(vector<int>& nums1, vector<int>& nums2, int k)
    {
        int res = 0, i = -1, j;
        while (++i < nums1.size())
        {
            j = -1;
            while (++j < nums2.size())
            {
                int mult = nums2[j] * k;
                if ( nums1[i] % mult == 0 )
                    ++res;
            }
        }
        return res;
    }
};

File: leet_3162_count_good.go
func numberOfPairs(nums1 []int, nums2 []int, k int) int {

    res := 0
    i := 0
    for i < len(nums1) {
        j := 0
        for j < len(nums2) {
            mult := nums2[j] * k
            if nums1[i] % mult == 0 {
                res++
            }
            j++
        }
        i++
    }
    return res
}
